pyface/ui/wx/util/event_loop_helper.py

```diff
@@ -0,0 +1,151 @@
+
+# (C) Copyright 2014-15 Enthought, Inc., Austin, TX
+# All right reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Code modified from pyface.ui.qt4.util.event_loop_helper
+import contextlib
+import threading
+
+import wx
+from traits.api import HasStrictTraits, Instance
+from pyface.api import GUI
+from pyface.util.guisupport import start_event_loop_wx
+
+
+class ConditionTimeoutError(RuntimeError):
+    pass
+
+
+@contextlib.contextmanager
+def dont_quit_when_last_window_closed(wx_app):
+    """
+    Suppress exit of the application when the last window is closed.
+
+    """
+    flag = wx_app.GetExitOnFrameDelete()
+    wx_app.SetExitOnFrameDelete(False)
+    try:
+        yield
+    finally:
+        wx_app.SetExitOnFrameDelete(flag)
+
+
+class EventLoopHelper(HasStrictTraits):
+    """ A helper class to provide basic event loop functionality. """
+
+    #: The gui toolkit application.
+    gui_app = Instance(wx.App)
+
+    #: The pyface gui implementation
+    gui = Instance(GUI, ())
+
+    #: Reference to a dummy wx event handler that will help bind events for
+    #: the timers.
+    _handler = Instance(wx.EvtHandler, ())
+
+    @contextlib.contextmanager
+    def event_loop(self, repeat=1):
+        """ Emulates an event loop `repeat` times.
+
+        Parameters
+        ----------
+        repeat : int
+            Number of times to process events. Default is 1.
+
+        """
+        yield

```

Comment:
Should this be in a try-finally block?  If there is an exception from within the context, we would still want to process the fired events right?

Reply:
Good point, I think a try-finally might be a good thing to do